pipenv/utils/resolver.py

```diff
@@ -447,9 +447,49 @@ def resolve(self):
                 self.resolved_tree.update(self.results)
         return self.resolved_tree
 
+    def _get_pipfile_markers(self, pipfile_entry):
+        sys_platform = pipfile_entry.get("sys_platform")
+        platform_machine = pipfile_entry.get("platform_machine")
+        markers = pipfile_entry.get("markers")
+
+        if sys_platform:
+            sys_platform = f"sys_platform {sys_platform}"
+        if platform_machine:
+            platform_machine = f"platform_machine {platform_machine}"
+
+        combined_markers = [
+            f"({marker})"
+            for marker in [sys_platform, markers, platform_machine]
```

Comment:
Nit but tuples take less RAM than lists.
```suggestion
            for marker in (sys_platform, markers, platform_machine)
```